compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionUtils.cpp

```diff
@@ -56,6 +63,118 @@ static bool canInsOperandTieWithOutsOperand(OpOperand *insOperand) {
   return llvm::any_of(linalgOp.getOutputOperands(), canTieWithOutsOperand);
 }
 
+/// Checks if a linalg op is a simple reduction of the innermost dimensions
+/// with identity map for the input.
+static bool isReductionOnInnermostDims(linalg::LinalgOp linalgOp) {
+  SmallVector<Operation *, 4> combinerOps;
+
+  auto numInputs = linalgOp.getNumInputs();
+  if (numInputs != 1 && numInputs != 2) return false;
+
+  if (linalgOp.getNumOutputs() != 1) return false;
+
+  // Check if the result dims are d0, d1, ..., which means the reduction is done
+  // in the innermost dimensions without an output transpose.
+  auto output = linalgOp.getOutputOperand(0);
+  auto outputIndexingMap = linalgOp.getTiedIndexingMap(output);
+  for (const auto &en : llvm::enumerate(outputIndexingMap.getResults())) {
+    auto expr = en.value();
+    if (auto dim = expr.dyn_cast<AffineDimExpr>()) {
+      if (dim.getPosition() != en.index()) return false;
+    } else {
+      return false;
+    }
+  }
+
+  if (!matchReduction(linalgOp.getRegionOutputArgs(), 0, combinerOps) ||
+      combinerOps.size() != 1) {
+    return false;
+  }
+
+  return true;
+}
+
+/// Check if the op is elementwise and the output indexing map is identity.
+static bool isSimpleElementwise(linalg::LinalgOp op) {
+  if (op.getNumLoops() != op.getNumParallelLoops()) return false;
+
+  if (!allIndexingsAreProjectedPermutation(op)) return false;
+
+  for (OpOperand *opOperand : op.getOutputOperands()) {
+    if (!op.getTiedIndexingMap(opOperand).isIdentity()) return false;
+  }
+  return linalg::hasOnlyScalarElementwiseOp(op->getRegion(0));
```

Comment:
At this level looking at region of Linalg ops should not be necessary. The indexing maps and iterator types should be sufficient for any constraints. So this constraint can be dropped as well

Reply:
> At this level looking at region of Linalg ops should not be necessary. The indexing maps and iterator types should be sufficient for any constraints. So this constraint can be dropped as well

So the part that we can't tell fro the indexing map is whether the reduction dim and parallel dim of elementwise have the same size. I guess we can ignore it for now but I think it will catch cases we might not want for GPU type of targets